python/lsst/metaserv/schemaToMeta.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2015 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+
+import commands
+import optparse
+import os
+import pprint
+import re
+import sys
+
+class SchemaToMeta(object):
+    """
+    SchemaToMeta class parses mysql schema file that can optionally contain
+    extra tokens in comments. Extracts information for each table:
+    * name
+    * engine
+    * per column:
+      - type
+      - notnull
+      - defaultValue
+      - <descr>...</descr>
+      - <unit>...</unit>
+      - <ucd>...</ucd>
+    and saves it in an array.
+
+    This code was originally written for schema browser
+    (in cat/bin/schema_to_metadata.py).
+    """
+
+    def __init__(self, inputFileName):
+        """
+        Constructor, prepares for execution.
+
+        @param inputFileName:   ASCII file containing the schem to be parsed
+        """
+        if not os.path.isfile(inputFileName):
+            sys.stderr.write("File '%s' does not exist\n" % inputFileName)
+            sys.exit(1)
+        self._inFName = inputFileName
+
+
+    def parse(self):
+        """Do actual parsing. Returns the retrieved structure as a table."""
```

Comment:
Can the structure of the returned table be described in more details. It looks a bit complex looking at the code below.